### ISSUE TITLE
cuda.Random_XorShift1024 test consistently fails on Power8+Pascal60 with cuda/9.2+gcc/7.2

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#3428]
This is occurring in nightly tests on the white testbed with various configurations of cuda/9.2+gcc/7.2 on Power8+Pascal60 hardware.

The test does not fail on the same hardware with cuda/10.1+gcc/7.4

I started a bisection on merge commits to track when this occurred (nightly testing was down for some time), it appears this was introduced with PR #3364 

Passing SHA: d13b4e7c8e1013c9760ebf0720334e721d9e6399
Failing SHA: b604752666fee805b6686a9e5310f50fa88dba5d

Sample configuration
```bash
module load cmake/3.12.3 cuda/9.2.88 gcc/7.2.0 ibm/xl/16.1.1

$KOKKOS_PATH/generate_makefile.bash --with-devices=Cuda,Serial --arch=Power8,Pascal60 --compiler=${KOKKOS_PATH}/bin/nvcc_wrapper --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="14" --ldflags="" --with-cuda=/home/projects/ppc64le-pwr8-nvidia/cuda/9.2.88 --kokkos-path=$KOKKOS_PATH --with-options= --with-cuda-options=  --no-examples
```

Failure output:
```
02:15:54 [ RUN      ] cuda.Random_XorShift1024
02:15:54 Test Seed:1601280938616796949
02:15:54 Test Scalar=int
02:15:54  -- Testing randomness properties
02:15:54 Pass: 0 1 -1 6.86779e-05 -0.000111521 || 0.000220971
02:15:54  -- Testing 1-D histogram
02:15:54 Density 1D: 2.47955e-07 0.00377728 -0.00525769 || 0.051031 10907 11766 || 11334.1 11377 || -72.6823 -0.823045
02:15:54  -- Testing 3-D histogram
02:15:54 Density 3D: 2.47955e-07 0.00489801 6.40393e-05 || 0.051031 1e+64 -1e+64
02:15:54 /home/jenkins/white/workspace/Kokkos_develop_white_cuda_92_gcc_720_pascal/kokkos/algorithms/unit_tests/TestRandom.hpp:407: Failure
02:15:54 Value of: 1
02:15:54 Expected: test_int.pass_mean
02:15:54 Which is: 0
02:15:54 Test Seed:1601280938651105052
02:15:54 Test Scalar=int
02:15:54  -- Testing randomness properties
02:15:54 Pass: 0 1 -1 0.000175569 6.12378e-05 || 0.000220971
02:15:54  -- Testing 1-D histogram
02:15:54 Density 1D: 2.47955e-07 0.00315355 -0.00399975 || 0.051031 10948 11782 || 11341.2 11377 || -55.2925 -0.823045
02:15:54  -- Testing 3-D histogram
02:15:54 Density 3D: 2.47955e-07 -0.00400319 0.000258702 || 0.051031 1e+64 -1e+64
02:15:54 /home/jenkins/white/workspace/Kokkos_develop_white_cuda_92_gcc_720_pascal/kokkos/algorithms/unit_tests/TestRandom.hpp:407: Failure
02:15:54 Value of: 1
02:15:54 Expected: test_int.pass_mean
02:15:54 Which is: 0
02:15:54 [  FAILED  ] cuda.Random_XorShift1024 (69 ms)
```

**Comment by crtrott** [comment]
I have been digging more into this, and it was not fixed by the suggested driver update ...

**Comment by crtrott** [comment]
This is the change which introduced the break:
https://github.com/kokkos/kokkos/compare/d13b4e7c8e1013c9760ebf0720334e721d9e6399..b604752666fee805b6686a9e5310f50fa88dba5d

**Comment by ndellingwood** [comment]
These nightlies are passing again thanks to fix in PR #3580 :)
